docs: migrate Spies wiki page to explainers

TG199 opened the pull request:
- Direct migration of wiki/Spies content to docs-next
- Part of #5248 wiki migration effort

<!-- 👋 Hi, thanks for sending a PR to mocha! 💖.
Please fill out all fields below and make sure each item is true and [x] checked.
Otherwise we may not be able to review your PR. -->

## PR Checklist

- [x] Addresses an existing open issue: works on #5248 (edit by @mark-wiemer)
- [x] That issue was marked as [`status: accepting prs`](https://github.com/mochajs/mocha/issues?q=is%3Aopen+is%3Aissue+label%3A%22status%3A+accepting+prs%22)
- [x] Steps in [CONTRIBUTING.md](https://github.com/mochajs/mocha/blob/main/.github/CONTRIBUTING.md) were taken

## Overview

<!-- Description of what is changed and how the code change does that. -->
This is a direct migration of the [Spies wiki page](https://github.com/mochajs/mocha/wiki/Spies) to `docs-next/src/content/docs/explainers/spies.mdx`.

- Added to docs-next (will add to legacy docs in follow-up)  
- Content placed in explainers section

mark-wiemer commented:
Thanks @TG199 ! I removed a dupe heading and added this new page to our nav menu for consistency, let me know if I made any crazy mistake! I'll leave this open for a few days to give others a chance to review as well :)

TG199 commented:
> Thanks @TG199 ! I removed a dupe heading and added this new page to our nav menu for consistency, let me know if I made any crazy mistake! I'll leave this open for a few days to give others a chance to review as well :)

okay

mark-wiemer commented:
"Compliance" check will fail but I'm OK with that, we don't want to close the corresponding bug yet, so I changed `fixes` to `works on` in the PR description

TG199 commented:
> "Compliance" check will fail but I'm OK with that, we don't want to close the corresponding bug yet, so I changed `fixes` to `works on` in the PR description

Okay
